packages/core/src/base/staticClient.ts

```diff
@@ -57,6 +81,15 @@ async function runFetch(
     ? _.clone(options.client.clientConfig.headers)
     : {};
 
+  // make-fetch-happen sets connection header to keep-alive by default which
+  // keeps node running unless it is explicitly killed.
+  // If the user wants to keep the connection alive they can set the Connection
+  // header to 'keep-alive' and we'll respect it. Otherwise, we set it to "close".
+  const connectionHeader = getHeader("connection", fetchOptions.headers);
```

Comment:
What about keeping close as the default and Users can override with keep alive?

Reply:
That's exactly what we are doing

Reply:
So we look for a connection header in the headers object provided, if there is already one we leave it alone otherwise we set it to "close"

Reply:
If it isn't very obvious I can add a little bit more detail in the comment to explain it better.

Reply:
Sorry I should have been more clear 
How about keeping it as close in clientConfig as default, and Users can override with keep-alive? The advantage is most of the changes in this method can be removed.

Reply:
that sounds like a good idea. Let me look into that.